File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int l = 0;
        int s = 0;
        for(int i=0;i<nums.size();i++){
            s = s+nums[i];
        }
        for(int i=0;i<nums.size();i++){
            int r = s- nums[i]-l;
            if (l==r)
                return i;
            l = l+nums[i];
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left = 0 
        right = sum(nums)

        for i in range(len(nums)):
            right = sum(nums[i+1:])
            if left == right:
                return i
            left += nums[i]
        return -1